examples/citi_bike/online_lp/launcher.py

```diff
@@ -0,0 +1,235 @@
+
+        while self._next_event_idx < len(finished_events):
+            # Calculate the interval index of this finished event.
+            interval_idx = (
+                finished_events[self._next_event_idx].tick // self._ticks_per_interval - self._num_recorded_interval
+            )
+            assert interval_idx >= 0, "The finished events are not sorted by tick!"
+            if interval_idx >= num_interval_to_record:
+                break
+
+            # Check the event_type to get the historical demand and supply.
+            event_type = finished_events[self._next_event_idx].event_type
+            if event_type == CitiBikeEvents.RequireBike:
+                # TODO: Replace it with a pre-defined PayLoad.
+                payload = finished_events[self._next_event_idx].payload
+                demand_history[interval_idx, payload.src_station] += 1
+            elif event_type == CitiBikeEvents.ReturnBike:
+                payload: BikeReturnPayload = finished_events[self._next_event_idx].payload
+                supply_history[interval_idx, payload.to_station_idx] += payload.number
+
+            # Update the index to the finished event that has not been processed.
+            self._next_event_idx += 1
+
+        self._num_recorded_interval += num_interval_to_record
+
+        # Record to the forecasters.
+        for i in range(self._num_station):
+            self._demand_forecaster[i].record(demand_history[:, i])
+            self._supply_forecaster[i].record(supply_history[:, i])
+
+    def _forecast_demand_and_supply(self) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Returns:
+            Tuple[np.ndarray, np.ndarray]:
+                The first item indicates the forecasting demand for each station in each time interval,
+                with shape: (num_time_interval, num_station).
+                The second item indicates the forecasting supply for each station in each time interval,
+                with shape: (num_time_interval, num_station).
+        """
+        demand = np.array(
+            [round(self._demand_forecaster[i].forecast()) for i in range(self._num_station)],
+            dtype=np.int32
+        ).reshape((1, -1)).repeat(self._num_time_interval, axis=0)
+
+        supply = np.array(
+            [round(self._supply_forecaster[i].forecast()) for i in range(self._num_station)],
+            dtype=np.int32
+        ).reshape((1, -1)).repeat(self._num_time_interval, axis=0)
+
+        return demand, supply
+
+    def __peep_at_the_future(self, env_tick:int):
+        demand = np.zeros((self._num_time_interval, self._num_station), dtype=np.int32)
+        supply = np.zeros((self._num_time_interval, self._num_station), dtype=np.int32)
+
+        for tick in range(env_tick, env_tick + self._num_time_interval * self._ticks_per_interval):
+            interval_idx = (tick - env_tick) // self._ticks_per_interval
```

Comment:
(env_tick, env_tick + plan_window_size)?

Reply:
I set num_time_interval=math.ceil(plan_window_size / decision_interval) in the code.

So that a plan window size that is not required to be divided by the decision interval.